### PR TITLE
Refactor: overload integers for number_t only

### PR DESCRIPTION
**Pull request by elazarg**
Trying to minimize the number of locations where accidental integer conversion may occur, and also simplify dsl_syntax. It is somewhat clunky; many explicit conversions may be avoided using templated constructor for e.g. interval_t.

This PR takes unimportant changes from #479.

### CODE REVIEW

File: src/crab/ebpf_domain.cpp [hunk 1]
```diff
@@ -607,13 +586,11 @@ static std::vector<linear_constraint_t> assume_unsigned_64bit_gt(const NumAbsDom
...
 
-static std::vector<linear_constraint_t>assume_unsigned_32bit_gt(const NumAbsDomain& inv, bool strict,
-                                                                variable_t left_svalue, variable_t left_uvalue,
-                                                                const crab::interval_t& left_interval_low,
-                                                                const crab::interval_t& left_interval_high,
-                                                                const linear_expression_t& right_svalue,
-                                                                const linear_expression_t& right_uvalue,
-                                                                const crab::interval_t& right_interval) {
+static std::vector<linear_constraint_t>
+assume_unsigned_32bit_gt(const NumAbsDomain& inv, bool strict, variable_t left_svalue, variable_t left_uvalue,
```
Comment (dthaler) [inline src/crab/ebpf_domain.cpp:590]: Why the line break after the type in this line and others above?  This looks inconsistent with 611, 694, etc.

Comment (elazarg) [reply src/crab/ebpf_domain.cpp:590]: No idea.
```suggestion
static std::vector<linear_constraint_t> assume_unsigned_32bit_gt(const NumAbsDomain& inv, bool strict, variable_t left_svalue, variable_t left_uvalue,
```

File: src/crab/ebpf_domain.cpp [hunk 2]
```diff
@@ -586,8 +586,7 @@ assume_unsigned_64bit_gt(const NumAbsDomain& inv, bool strict, variable_t left_s
     }
 }
 
-static std::vector<linear_constraint_t>
-assume_unsigned_32bit_gt(const NumAbsDomain& inv, bool strict, variable_t left_svalue, variable_t left_uvalue,
+static std::vector<linear_constraint_t> assume_unsigned_32bit_gt(const NumAbsDomain& inv, bool strict, variable_t left_svalue, variable_t left_uvalue,
```
Comment (dthaler) [inline src/crab/ebpf_domain.cpp:589]: This change was done in this line but there were a bunch of other similar line breaks added in the PR too

Comment (elazarg) [reply src/crab/ebpf_domain.cpp:589]: I see

Comment (elazarg) [reply src/crab/ebpf_domain.cpp:589]: Okay this is just how AlwaysBreakAfterReturnType=None option works in clang-format for PenaltyReturnTypeOnItsOwnLine=60 as is default.

(I'm waiting for clang-format 17 to always enforce clang-formatting, but it should be useful to run it anyway, when there are no C++-style attributes).

**Pull request merged**